File: src/java/ironfist/ui/jcurses/Runner.java
package ironfist.ui.jcurses;

import ironfist.Creature;
import ironfist.Referee;

/**
 * DOCUMENT ME!
 * 
 * @author pmgremo
 */
public class Runner implements Runnable {

  private JCursesClient client;
  private Creature creature;

  /**
   * Creates a new Runner object.
   * 
   * @param client
   *          DOCUMENT ME!
   * @param creature
   *          DOCUMENT ME!
   */
  public Runner(JCursesClient client, Creature creature) {
    this.client = client;
    this.creature = creature;
  }

  /**
   * @see java.lang.Runnable#run()
   */
  public void run() {
    Referee.run(creature);
    client.setGameEnd();
  }

  /**
   * DOCUMENT ME!
   */
  public void stop() {
    Referee.stop();
  }
}

File: src/main/java/ukalus/ui/jcurses/Runner.kt
package ukalus.ui.jcurses

import ukalus.Creature
import ukalus.Referee

/**
 * DOCUMENT ME!

 * @author pmgremo
 */
class Runner
/**
 * Creates a new Runner object.

 * @param client
 * *          DOCUMENT ME!
 * *
 * @param creature
 * *          DOCUMENT ME!
 */
(private val client: JCursesClient, private val creature: Creature) : Runnable {

    /**
     * @see java.lang.Runnable.run
     */
    override fun run() {
        Referee.run(creature)
        client.setGameEnd()
    }

    /**
     * DOCUMENT ME!
     */
    fun stop() {
        Referee.stop()
    }
}